tests/test/extern_types.rs

```diff
@@ -0,0 +1,72 @@
+//! Tests for extern types
+
+use super::*;
+
+#[test]
+fn extern_ty_trait_impl() {
+    test! {
+        program {
+            extern { type A; }
+            trait Foo {}
+            impl Foo for A {}
+        }
+
+        goal { A: Foo } yields { "Unique" }
+    }
+}
+
+#[test]
+fn extern_ty_single() {
+    lowering_success! {
+        program {
+            extern { type A; }
+        }
+    }
+}
+
```

Comment:
Can you provide like a one-line comment on each of these for *why* this test exists (and passes as it is).
Honestly, it's probably:
`extern types are always well-formed`
and
`extern types don't implement any builtin traits`